### DISCUSSION TITLE
Debugging @events.autovox_policy

### DISCUSSION THREAD

**Discussion by Serrindipity** [xonsh/xonsh#5704] [Q&A]
I've defined an autovox policy as documented [here](https://xon.sh/python_virtual_environments.html) in my `.xonshrc` file.
```xonsh
@events.autovox_policy
def venvsearch_policy(path, **_):
# Searches through venvs in $VIRTUALENV_HOME and returns the path of the one (if any) that is linked (using venv/.project)
    for v in p`$VIRTUALENV_HOME/.*/.project`:
		if $(cat @(v)) == str(path) and v.exists():
			print("Syncing venv")
			return v
```

When I `cd` to a directory with a linked project, `Syncing venv` gets printed, but the virtual environment doesn't get activated. What's the best way to debug something like this? Is is a problem with `vox` / `autovox`?

**Comment by anki-code** [comment]
The implementation of this [event is here](https://github.com/xonsh/xontrib-vox/blob/fe51b7b9a52adbf94ce0a22f824db6d3455b63f7/xontrib/autovox.py#L42-L46). Check that [activation is working here](https://github.com/xonsh/xontrib-vox/blob/fe51b7b9a52adbf94ce0a22f824db6d3455b63f7/xontrib/autovox.py#L61-L76). 
And may be @yaxollum can help as an expert in venv activator :)
